packages/react-native/ReactCommon/react/renderer/components/textinput/platform/ios/react/renderer/components/iostextinput/TextInputShadowNode.cpp

```diff
@@ -38,6 +46,14 @@ TextInputShadowNode::TextInputShadowNode(
   }
 }
 
+void TextInputShadowNode::initialize() {
+  if (yogaNode_.style().display() == yoga::Display::Contents) {
+    traits_.set(ShadowNodeTraits::Trait::Hidden);
+  } else {
+    traits_.unset(ShadowNodeTraits::Trait::Hidden);
+  }
+}
```

Comment:
The same approach wasn't working on Android, I assume that it may be related to this https://github.com/facebook/react-native/blob/151008f8c209eb9f930668343e8a143bd41b0353/packages/react-native/ReactCommon/react/renderer/mounting/Differentiator.cpp#L217-L223